services/spar/src/Spar/Intra/Brig.hs

```diff
@@ -118,6 +119,17 @@ confirmUserId buid = do
   maybe (pure Nothing) (const . pure . Just $ buid) (Brig.userTeam =<< usr)
 
 
+assertIsTeamOwner :: (HasCallStack, MonadError ServantErr m, MonadSparToBrig m) => Maybe UserId -> TeamId -> m ()
+assertIsTeamOwner mbuid tid = do
+  let reject = throwError err403 { errBody = encode [aesonQQ|{"error":"you need to be team admin to create an IdP"}|] }
```

Comment:
Instead of returning errors that are `{"error": "simple strings"}`, could you follow the error structure used in other services, that is to say, include a `code` and a descriptive, stable `label` (the idea being that clients can pattern match on the `label` field which should never change once introduced, and we can still change the exact textual error description while remaining backwards-compatible). Example:

```
{
  "code": 400,
  "message": "[path] 'id' invalid: Failed reading: Invalid UUID",
  "label": "client-error"
}
```

See e.g. https://github.com/wireapp/wire-server/blob/develop/services/galley/src/Galley/API/Error.hs#L45 for some more examples.